app/logic/users.py

```diff
@@ -10,28 +12,58 @@ def isEligibleForProgram(program, user):
     :param user: accepts a User object or userid
     :return: True if the user is not banned and meets the requirements, and False otherwise
     """
-
-    if (ProgramBan.select().where(ProgramBan.user == user, ProgramBan.program == program).exists()):
+    now = datetime.datetime.now()
+    if (ProgramBan.select().where(ProgramBan.user == user, ProgramBan.program == program, ProgramBan.endDate > now).exists()):
         return False
 
     return True
 
-def addRemoveInterest(rule, program_id, currentUser):
+def addRemoveInterest(rule, program_id, username):
```

Comment:
Make this two functions. The controller should worry about whether or not "add" or "delete" is in the rule, and then call the appropriate function.

Reply:
Fixed!